### PR TITLE
Added auto foraging of bushes and trees

### LINKED ISSUE #24438: Need auto forage mode to reduce tedium for foraging
Game version:
master
Operating system:
any
Tiles or curses:
any
Mods active:
any
### Expected behavior
We have auto pickup, auto pulping but no auto forage mode.

When active, auto forage mode should like what auto pulping mode does except foraging nearby trees/underbrush.
### Actual behavior
Many Many Many key press.

### Steps to reproduce the behavior
 Try forage in your nearest forest.

### LINKED ISSUE #25300: Walking on a road while wielding a pickaxe should not destory the road
When you're wielding a pickaxe or a similar tool and you attempt to walk on a road (a pavement tile), you "mine" the tile instead of walking.

What makes this worse is that it also destroys any vehicle tile there. So, instead of opening a car door or getting into a car, you instead destroy the car door if it's on the road and you're wielding a pickaxe.

**To Reproduce**  

0. Make you sure you have automatic mining enabled (it should be on by default).
1. Wish for and wield a pickaxe.
2. Try to walk on a road.

**Expected behavior**  

No roads or vehicles are destroyed by just walking on them.

**Versions and configuration(please complete the following information):**
 - Game Version: 0.C-31400-gdceede2

### PR DESCRIPTION
**Pull request by OzoneH3**
#### Summary
```SUMMARY: Features "Added auto foraging of bushes and trees"```

#### Purpose of change
Fixes #24438
Fixes #25300

#### Additional context
Moved auto mining under "Additional auto features" and set it to false by default.

### CODE REVIEW

**Comment by cake-pie** [comment]
While this does ameliorate #25300 by leaving auto mining off by default, I'm not sure it should be considered a complete fix that closes the issue. A player who turned on auto mine is still possibly going to get a nasty surprise in the shape of

https://github.com/CleverRaven/Cataclysm-DDA/issues/25300#issuecomment-417953273
> when you've noticed that this is an issue ... your car might already be destroyed beyond repair due to this behavior.

**Comment by OzoneH3** [comment]
You'd have to enable both Additional auto features and auto mining without reading any descriptions for this to be a real problem. Can't help them any more than that.

File: src/game.cpp [hunk 1]
```diff
@@ -10723,9 +10726,33 @@ void game::place_player( const tripoint &dest_loc )
...
 
-    //Auto pulp or butcher
-    if( get_option<bool>( "AUTO_PULP_BUTCHER" ) && mostseen == 0 ) {
-        const std::string pulp_butcher = get_option<std::string>( "AUTO_PULP_BUTCHER_ACTION" );
+    //Auto pulp or butcher and Auto foraging
+    if( get_option<bool>( "AUTO_FEATURES" ) && mostseen == 0 ) {
+        static const direction adjacentDir[8] = { NORTH, NORTHEAST, EAST, SOUTHEAST, SOUTH, SOUTHWEST, WEST, NORTHWEST };
+
+        const std::string forage_type = get_option<std::string>( "AUTO_FORAGING" );
+        if( forage_type != "off" ) {
```
Comment (ZhilkinSerg) [inline src/game.cpp]: A little bit late, but I guess we also need to add `!forage_type.empty() && ` here.

Comment (ZhilkinSerg) [reply src/game.cpp]: Maybe you are right.

I just run into some issues with `forage_type` which was set to NULL and caused crash several lines later while testing another unrelated PR.

Comment (ZhilkinSerg) [reply src/game.cpp]: See - appears after I debug teleport to some place:

image

**Pull request merged**